### PR TITLE
Add check for regex when isPattern is true

### LINKED ISSUE #968: Subscription using wrong regex patterns
Raised during Developers Week Viena 2015.

It seem there are problems with subscriptions using patterns when the pattern is not a righ regex. Eg, when using `*` (wrong) instead of `.*` (right). Details below (IPs has been offuscated).

Orion Context Broker GE
version 0.14.0
image 3.4

Subscription request

Endpoint: http://A.B.C.D:1026/ngsi10/subscribeContext

payload:

```
{
    "entities": [
        {
            "type": "Patient",
            "isPattern": "true",
            "id": "*"
        }
    ],
    "attributes": [
        "Blood glucose",
        "Body temperature",
        "Systolic blood pressure",
        "Diastolic blood pressure",
        "Respiration rate",
        "Heart rate",
        "Blood oxygen",
        "timestamp"
    ],
    "reference": "http://A.B.C.D:5999/orion2mysql",
    "duration": "P1Y",
    "notifyConditions": [
        {
            "type": "ONCHANGE",
            "condValues": [
                "timestamp"
            ]
        }
    ],
    "throttling": "PT1S"
}
```

Update request:
Endpoint: http://A.B.C.D:1026/ngsi10/updateContext

Payload:
```
{
     "contextElements":
     [
        {
            "type": "Patient",
            "isPattern": "false",
            "id": "Patient138866191",
            "attributes": [
            {
                "name": "Heart rate",
                "type": "bpm",
                "value": "72"
            },
            {
                "name" : "Diastolic blood pressure",
                "type" : "mmHg",
                "value" : "11.6"
            },
            {
                "name" : "Systolic blood pressure",
                "type" : "mmHg",
                "value" : "7.6"
            },
            {
                "name" : "timestamp",
                "type" : "Europe/Athens",
                "value" : "2014-12-09 17:05:34"
            }
            ]
        }
    ],
    "updateAction": "APPEND"
}
```

You have to replace the id:"Patient138866191" included in the update request payload with  id:"GR138866191" to get a right response.

### PR DESCRIPTION
**Pull request by crbrox**
Fixes #968

@fgalan 
@kzangeli 

Also
change all regex to use extended regular expressions
add idPattern check in parseEntityObject

### CODE REVIEW

**Comment by kzangeli** [comment]
LGTM, but with a comment - some day we need to forward the composed regex instead of creating it anew.
Also ... CNR?

File: test/functionalTest/cases/0968_ngsi10_asterisk_subs/0968_ngsi10_asterisk_subs.test [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            "condValues": [
+                "timestamp"
+            ]
+        }
+    ],
+    "throttling": "PT1S"
+}
+'
+
+orionCurl --url /ngsi10/subscribeContext --payload "$payload"
```
Comment (fgalan) [inline test/functionalTest/cases/0968_ngsi10_asterisk_subs/0968_ngsi10_asterisk_subs.test]: /ngsi10 -> /v1

Comment (crbrox) [inline test/functionalTest/cases/0968_ngsi10_asterisk_subs/0968_ngsi10_asterisk_subs.test]: Fixed 1a2f351

File: src/lib/cache/subCache.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ EntityInfo::EntityInfo(const std::string& _entityId, const std::string& _entityT
   {
     // FIXME P5: recomp error should be captured? have a look to other usages of regcomp()
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: Suggested CNR entry:

"Fix: wrong regex patterns detection (#968)"

Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: wrong => invalid

Comment (crbrox) [inline src/lib/cache/subCache.cpp]: fixed 5b15ae0

**Comment by fgalan** [comment]
LGTM

**Comment by kzangeli** [comment]
LGTM (again)

**Pull request merged**